src/pages/Portfolio/ImpactDashboard/YourDonations.vue

```diff
@@ -0,0 +1,277 @@
+					query: gql`query yourDonations ($filters: TransactionSearchFilters, $limit: Int) {
+						my {
+							id
+							transactions(filter: $filters, limit: $limit) {
+								totalCount
+								values {
+									createTime
+									amount
+								}
+							}
+							userStats {
+								amount_donated
+								amount_of_loans
+							}
+						}
+					}`,
+					variables: {
+						filters: transactionFilters,
+						limit: this.limit,
+					},
+				}).then(({ data }) => {
+					this.loading = false;
+					const currentYear = this.currentDate.getFullYear();
+					const transactions = data?.my?.transactions?.values ?? [];
+					const amountOfLoans = numeral(data?.my?.userStats?.amount_of_loans ?? 0);
+					const donation = numeral(data?.my?.userStats?.amount_donated ?? 0);
+					if (amountOfLoans.value() > 0) {
+						donation.multiply(1 / amountOfLoans.value()).multiply(25);
+					}
+					this.averageDonation = donation.format('$0,0[.]00');
+
+					this.totalDonations = data?.my?.transactions?.totalCount ?? 0;
+					this.currentYearDonations = this.filterAndSumDonations(transactions, currentYear);
+					this.lastYearDonations = this.filterAndSumDonations(transactions, currentYear - 1);
+
+					if (this.totalDonations > this.limit) {
+						let offset = this.limit;
+						while (offset < this.totalDonations) {
+							this.getRemainDonations(transactionFilters, offset)
+								.then(donations => {
+									const remainDonations = donations?.data?.my?.transactions?.values ?? [];
+									this.currentYearDonations += this.filterAndSumDonations(
+										remainDonations, currentYear
+									);
+									this.lastYearDonations += this.filterAndSumDonations(
+										remainDonations, currentYear - 1
+									);
+								});
+
+							offset += this.limit;
+						}
+					}
+				}).finally(() => {
+					this.loadingPromise = null;
+				});
+			}
+		},
+		whenVisible() {
+			this.fetchAsyncData();
+		},
+		getRemainDonations(transactionFilters, offset) {
+			return this.apollo.query({
+				query: gql`query yourDonations ($filters: TransactionSearchFilters, $limit: Int, $offset: Int) {
+						my {
+							id
+							transactions(filter: $filters, limit: $limit, offset: $offset) {
+								values {
+									createTime
+									amount
+								}
+							}
+						}
+					}`,
+				variables: {
+					filters: transactionFilters,
+					limit: this.limit,
+					offset,
+				},
+			});
+		},
+		filterAndSumDonations(donations, targetYear) {
+			return donations.filter(donation => {
+				const donationYear = new Date(donation.createTime).getFullYear();
+				return donationYear === targetYear;
+			}).reduce((acc, donation) => {
+				return acc + donation.amount * -1;
+			}, 0);
+		}
+	},
+	filters: {
+		dollarFormat(value) {
+			return numeral(value).format('$0,0[.]00');
+		}
+	}
+};
+</script>
+
+<style lang="postcss" scoped>
+.stat-placeholder {
+	margin-bottom: 6px;
+	@apply tw-mt-1 tw-h-4.5;
+
+	@screen md {
```

Comment:
Do these nested `@screen` usages play nicely with postcss? I know we don't have the nesting module, but I wasn't sure if `@screen` counted as nesting here.